providers/base/units/usb/usb-c.pxu

```diff
@@ -208,3 +208,129 @@ _steps:
  2. Start the test
  3. When the message "INSERT NOW" is shown, plug in the adapter to Type-C port
 estimated_duration: 20
+
+id: usb-c-otg/g_serial
+_summary: Check if USB OTG can work as a serial port.
+category_id: com.canonical.plainbox::usb
+plugin: user-interact-verify
+user: root
+estimated_duration: 300.0
+command:
+   modprobe g_serial
+_steps:
+ 1. Press enter to probe USB OTG g_serial
+ 2. Connect a USB cable from an Ubuntu host to the USB OTG (it's the USB type C connector) port on DUT.
+ 3. On the host, to check if ttyACM0 has been probed
+    $ ls /dev/ttyACM0
+ 4. On the host, listen to the ttyACM0 port as a receive side.
+    $ sudo su
+    $ cat /dev/ttyACM0
+ 5. On DUT, send a string via /dev/ttyGS0 as a send side.
+    $ sudo su
+    $ echo 123 > /dev/ttyGS0
+ 6. Check if received the string "123" on the host side.
+ 7. Check the string sending from the host to DUT by repeating steps 4-6, but swap the send and receive sides.
+_verification:
+    Does string send and receive work?
+flags: preserve-locale also-after-suspend
+imports: from com.canonical.plainbox import manifest
+requires: manifest.has_usbc_otg == 'True'
+
+id: usb-c-otg/g_serial-cleanup
+_summary: Cleanup USB OTG serial interface setup after serial device test
+plugin: shell
+after: usb-c-otg/g_serial
+category_id: com.canonical.plainbox::usb
+command:
+  echo "Removing g_serial module..."
+  modprobe -r g_serial
+user: root
+estimated_duration: 1s
+flags: also-after-suspend
```

Comment:
I think this might not work to clean up `after-suspend-usb-c-otg/g_serial`. It's going to create a `after-suspend-usb-c-otg/g_serial-cleanup` that's gonna be run after `usb-c-otg/g_serial`, which is not what you want.

You will probably have to use the `siblings` field to manually set the `after` field to `after-suspend-usb-c-otg/g_serial`. You will also have to manually set all the stuff that the `also-after-suspend` flag usually does for you. Check the [Checkbox job unit documentation](https://checkbox.readthedocs.io/en/latest/units/job.html) for more info about the `siblings` field and what the `also-after-suspend` flag does.